test/puppetlabs/trapperkeeper/services_test.clj

```diff
@@ -62,6 +64,67 @@
           app (bootstrap-services-with-empty-config [service1])]
       (is (not (nil? app))))))
 
+(deftest test-restart-file-correct-count
+  (testing "create a restart file and check that it correctly increments when a service starts"
+    (let [service1 (service Service1
+                            []
+                            (service1-fn [this] "hi"))
+          temp-file (kitchensink/temp-file-name "counter")]
+          (with-app-with-config app [service1] {:global {:restart-file temp-file}}
+            (app/restart app)
+            (is (= (slurp temp-file) "2"))))))
+
+(deftest test-restart-file-HUP-restart
```

Comment:
shoot, that (app/restart app) should not be there. sorry a bit out of a it this morning